src/App.tsx

```diff
@@ -4,57 +4,110 @@
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
+import * as userService from './api/users';
+import * as postService from './api/posts';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { Post } from './types/Post';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isPostsLoading, setIsPostsLoading] = useState(false);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [hasError, setHasError] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    userService
+      .getUsers()
+      .then(setUsers)
+      .catch(() => setHasError(true));
+  }, []);
 
-              <Loader />
+  useEffect(() => {
+    setIsPostsLoading(true);
+    postService
+      .getUserPosts(selectedUser?.id || 0)
+      .then(setUserPosts)
+      .catch(() => setHasError(true))
+      .finally(() => setIsPostsLoading(false));
+  }, [selectedUser]);
```

Comment:
The dependency array for `useEffect` should include `selectedUser?.id` instead of `selectedUser` to avoid unnecessary re-fetching when other properties of `selectedUser` change.